packages/mobile/src/components/premium-track-purchase-drawer/PremiumTrackPurchaseDrawer.tsx

```diff
@@ -0,0 +1,182 @@
+import { useCallback } from 'react'
+
+import {
+  cacheTracksSelectors,
+  cacheUsersSelectors,
+  formatUSDCWeiToUSDString,
+  isPremiumContentUSDCPurchaseGated
+} from '@audius/common'
+import { View } from 'react-native'
+import { useSelector } from 'react-redux'
+
+import IconCart from 'app/assets/images/iconCart.svg'
+import { LockedStatusBadge, Text, Button } from 'app/components/core'
+import { NativeDrawer } from 'app/components/drawer'
+import { useDrawer } from 'app/hooks/useDrawer'
+import { makeStyles, flexRowCentered } from 'app/styles'
+import { useThemeColors } from 'app/utils/theme'
+
+import { TrackDetailsTile } from '../track-details-tile'
+
+const { getUser } = cacheUsersSelectors
+const { getTrack } = cacheTracksSelectors
+
+const PREMIUM_TRACK_PURCHASE_MODAL_NAME = 'PremiumTrackPurchase'
+
+const messages = {
+  title: 'Complete Purchase',
+  summary: 'Summary',
+  artistCut: 'Artist Cut',
+  audiusCut: 'Audius Cut',
+  always0: 'Always $0',
```

Comment:
I don't know why, but I always feel like numbers in variable names is an anti-pattern. Maybe because you can't start a variable with a number in some languages? Anyway, my minor nit is that `alwaysZero` reads a little easier.